google/cloud/opentelemetry/internal/time_series.cc

```diff
@@ -51,6 +86,36 @@ google::api::Metric ToMetric(
   return proto;
 }
 
+google::monitoring::v3::TimeInterval ToNonEmptyTimeInterval(
+    opentelemetry::sdk::metrics::MetricData const& metric_data) {
+  // GCM requires that time intervals are non-empty. To achieve this, we
+  // override the end value to be at least 1ms after the start value.
+  // https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/babed4870546b78cee69606726961cfd20cbea42/exporter/metric/metric.go#L604-L609
```

Comment:
non-empty does not require a full millisecond, just one nanosecond would do.  I think you would be better quoting:

https://cloud.google.com/monitoring/api/ref_v3/rpc/google.monitoring.v3#timeinterval

where it says that a millisecond is needed for non-GAUGE metrics.  Note that for GAUGE metrics the start time is ignored so you would be adjusting the thing that is actually used.

Seems like for GAUGE metrics you would want to leave the interval alone, and only adjust the end metric for non-GAUGE metrics.

Reply:
> I think you would be better quoting...

Done, and `s/NonEmpty/NonGauge/`. Thanks.

> Note that for GAUGE metrics the start time is ignored so you would be adjusting the thing that is actually used.

I am aware. The conversion for gauge metrics won't use this function.

I hear you on having a fake start vs. having a fake end. I am going to do what the other languages do, though.

> Seems like for GAUGE metrics you would want to leave the interval alone, and only adjust the end metric for non-GAUGE metrics.

Yes. That is the plan.